### ISSUE TITLE
Can I put other classes and directives in a codeBlockCaptions marker?

### ISSUE THREAD

**Issue by colindean** [lierdakil/pandoc-crossref#269]
I'm trying to use `pandoc-crossref` and `pandoc-include-code` together.

With this in `test.md`:

    # This is a test

    Listing: ruby.rb {#lst:ruby include="ruby.rb"}

    ```ruby
    puts "Hello world"
    ```

and this in `ruby.rb`

```ruby
puts "Hello world from ruby.rb"
```
when compiled with 

    pandoc test.md -o test.pdf --filter pandoc-crossref --filter pandoc-include-code -M codeBlockCaptions=true

I'm seeing this:

![Screen Shot 2020-06-29 at 16 32 19](https://user-images.githubusercontent.com/197224/86053181-2033a300-ba26-11ea-9f35-06b84dc95656.png)

When I remove the `include="ruby.rb"`, then I see a sensible output:

![Screen Shot 2020-06-29 at 16 33 01](https://user-images.githubusercontent.com/197224/86053317-5113d800-ba26-11ea-868a-af83c7bc22b7.png)

When I put _this_ into `test.md`:

     # This is a test

     Listing: ruby.rb {#lst:ruby}

     ```{.ruby include="ruby.rb"}
     puts "Hello world"
    ```

    See listing @lst:ruby

I see what I expect, but I'd really prefer to put the include on the listing line.

![Screen Shot 2020-06-29 at 16 35 48](https://user-images.githubusercontent.com/197224/86053663-d0a1a700-ba26-11ea-9ef7-b0f725a8f9ed.png)

My goal is not to have to put anything on the triple-backtick line other than the language in use. My editor highlights those code blocks nicely but when I add anything else to the line, like `caption` or `include` then that highlighting no longer happens.

I would expect these classes to apply to the codeblock when specified in the class list in the Listing read by `codeBlockCaptions`.

**Comment by lierdakil** [comment]
```
Listing: ruby.rb {#lst:ruby include="ruby.rb"}
```

This won't (can't) work. Long story short, this is a giant hack that re-parses bits of Pandoc's AST, and the complexity of parsing arbitrary attributes would just shoot straight through the roof. The only thing parsed in captions is the identifier.

It would probably be easier to find/write a more robust highlighting grammar for your editor.

But also, side note: why do you even care about "nice highlighting" when using pandoc-include-code? I thought the point of this filter is to replace the code block content with some external file?

**Comment by colindean** [comment]
Thanks for the explanation. That makes sense.

Eh, using pandoc-include-code in that example was a bad example. I’d put .numberedLines and friends in there with the same output effect/problem. In the doc I’m building, I’ve got a mix of included code and embedded code.

**Comment by lierdakil** [comment]
Huh. Not sure if this is still relevant, but there's actually a reasonably straightforward workaround:

`````markdown
:::{#lst:code .listing include="ruby.rb"}
Listing caption
```ruby
puts "Hello world"
```
:::
`````

Now, this doesn't actually work as of right now, but it's simple enough to _make_ it work. Let me know if you're still interested.

Don't know why I didn't think of this sooner. Probably was either busy or tired when I answered and then the issue just gut buried =\ That happens. Sorry.

**Comment by lierdakil** [comment]
Released in v0.3.10.0.
